src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -267,7 +308,8 @@ abstract class AbstractCheckTask(
                     .toProto(parentEventID))
                 .build())
         } finally {
-            sequenceSubject.onSuccess(Legacy(executorService, lastSequence))
+            val untrustedForFutureTask = !bufferContainsStartMessage && !hasMessagesInTimeoutInterval
```

Comment:
Maybe `!hasMessagesInTimeoutInterval` will be enough here

Reply:
But shouldn't we check that there are no messages in the specified interval before we passed untrusted flag?

Reply:
If the `!hasMessagesInTimeoutInterval` then we didn't find any message that has a timestamp less than the timestamp in the checkpoint. If the message from the checkpoint was in the buffer we would found it and the `hasMessagesInTimeoutInterval` would be `true`. If I am not mistaking of course

Reply:
Message timestamp timeout is checkpoint timestamp + message timeout. If the buffer missed the checkpoint message and the next message following it that falls in this interval, then we will get `bufferContainsStartMessage = false` and `hasMessagesInTimeoutInterval = true`

Reply:
> @OptimumOpium So if we have no messages in the interval or a missed start message from the previous task, should we still need to execute the task? In other words, is there no reason not to execute the task?

In my opinion, yes. We should execute the check rule event if it is started from an untrusted checkpoint. If you have doubts about that we can add a parameter that will configure the check1 to behave the way we need (fail the rule if it is started from the untrusted checkpoint without any execution or try to execute the next rule and add the failed event to the result regardless the rule succeeds or not).

Reply:
And what? In this case, the checkpoint is not untrusted and the `!hasMessagesInTimeoutInterval` will return `false` (as it should)